pennylane/transforms/decompose.py

```diff
@@ -98,12 +152,31 @@ def __init__(self, gate_set=None, max_expansion=None):
                 gate_set = set([gate_set])
 
             if isinstance(gate_set, Iterable):
-                gate_types = tuple(gate for gate in gate_set if isinstance(gate, type))
-                gate_names = set(gate for gate in gate_set if isinstance(gate, str))
-                self.gate_set = lambda op: (op.name in gate_names) or isinstance(op, gate_types)
+                target_gate_types = tuple(gate for gate in gate_set if isinstance(gate, type))
+                target_gate_names = set(gate for gate in gate_set if isinstance(gate, str))
+                self.gate_set = lambda op: (op.name in target_gate_names) or isinstance(
+                    op, target_gate_types
+                )
+
+                if qml.decomposition.enabled_graph():
+
+                    if target_gate_types:
+                        raise TypeError(
+                            f"The graph-based decomposition doesn't support Operator types: {target_gate_types}."
+                        )
+
+                    self._target_gate_names = target_gate_names | set(
+                        gate.name for gate in target_gate_types
+                    )
+
             else:
                 self.gate_set = gate_set
 
+                if qml.decomposition.enabled_graph():
+                    raise TypeError(
```

Comment:
Since line#173 is covered, I guess we only need to find the corresponding test suite that has a Callable `gate_set` and make its `enabled_graph` counterpart to cover line#176